build-tools/packages/build-tools/src/repoPolicyCheck/handlers/npmPackages.ts

```diff
@@ -140,6 +154,32 @@ function getReadmeInfo(dir: string): IReadmeInfo {
     };
 }
 
+function ensurePrivatePackagesComputed(): void {
+    if (privatePackages) {
+        return;
+    }
+
+    privatePackages = new Set();
+    const pathToGitRoot = child_process.execSync("git rev-parse --show-cdup", { encoding: "utf8" }).trim();
+    const p = child_process.spawn("git", ["ls-files", "-co", "--exclude-standard", "--full-name", "**/package.json"]);
+    const lineReader = readline.createInterface({
+        input: p.stdout,
+        terminal: false
+    });
+
+    lineReader.on('line', line => {
+        const filePath = path.join(pathToGitRoot, line).trim().replace(/\\/g, "/");
+        if (fs.existsSync(filePath)) {
+            const packageJson = JSON.parse(readFile(filePath));
+            if (packageJson.private) {
+                privatePackages.add(packageJson.name);
+            }
+        }
```

Comment:
`git ls-files` will include files that exist in the index but not on disk. No-oping in that case is reasonable enough, since leaving dangling references to deleted packages will fail the install step